flume-ng-auth/src/test/java/org/apache/flume/auth/TestFlumeAuthenticator.java

```diff
@@ -110,6 +111,46 @@ public void testFlumeLogin() throws IOException {
     }
   }
 
+  /**
+   * Test whether the exception raised in the <code>PrivilegedExceptionAction</code> gets
+   * propagated as-is from {@link KerberosAuthenticator#execute(PrivilegedExceptionAction)}.
+   */
+  @Test(expected = IOException.class)
+  public void testKerberosAuthenticatorExceptionInExecute() throws Exception {
+    String principal = flumePrincipal;
+    String keytab = flumeKeytab.getAbsolutePath();
+
+    // Clear the previous statically stored logged in credentials
+    FlumeAuthenticationUtil.clearCredentials();
```

Comment:
I think this cleanup should be part of tearDown. (Here and in other tests as well.)